src/freenet/client/async/SplitFileInserter.java

```diff
@@ -139,7 +143,9 @@ public void onResume(ClientContext context) throws InsertException, ResumeFailed
                     context.getChkInsertScheduler(realTime).fetchingKeys(), context.persistentFG, 
                     context.persistentFileTracker, context.getPersistentMasterSecret());
             storage.onResume(context);
-            this.sender = new SplitFileInserterSender(this, storage);
+            synchronized(this) {
+                this.sender = new SplitFileInserterSender(this, storage);
```

Comment:
Because it can be accessed from multiple threads. However, it is only set on starting a new request and in onResume() (called when restarting a request after serialising in from client.dat). It sadly can't be final, but it won't change once it's set.